app/controllers/chats_controller.rb

```diff
@@ -7,11 +7,10 @@ def create
     @sender = current_user
     @recipient = User.find_by(email: chat_params[:email])
     if Chat.between(@sender.id, @recipient.id).present?
-      @chat = Chat.between(@sender.id, @recipient.id)
+      @chat = Chat.between(@sender.id, @recipient.id).first
     else
       @chat = Chat.create({sender_id: @sender.id, recipient_id: @recipient.id})
     end
```

Comment:
Could you push this logic down to the model? Maybe a create_or_retrieve method on the class?